task/mgmt-gateway/src/update/sp.rs

```diff
@@ -11,19 +12,54 @@ use gateway_messages::{
     UpdateInProgressStatus, UpdateStatus,
 };
 
+#[cfg_attr(not(feature = "auxflash"), path = "sp/stub_auxflash.rs")]
+mod auxflash;
+
+use self::auxflash::ChckScanResult;
+use self::auxflash::IngestDataResult as AuxFlashIngestDataResult;
+use self::auxflash::State as AuxFlashState;
+
+cfg_if! {
+    if #[cfg(feature = "auxflash")] {
+        use drv_auxflash_api::AuxFlash;
+
+        userlib::task_slot!(AUX_FLASH_SERVER, auxflash);
+    } else {
+        // If the auxflash feature isn't enabled, we can stub out a fake
+        // `AuxFlash` server that we never actually do anything with (other than
+        // construct it).
+        use userlib::TaskId;
+
+        struct AuxFlash;
+        impl From<TaskId> for AuxFlash {
+            fn from(_: TaskId) -> Self {
+                Self
+            }
+        }
+
+        struct FakeAuxFlashServer;
```

Comment:
Nit: maybe name `FakeAuxFlashTaskSlot`, since that's a more accurate description of this type.